packages/git-safe-dependencies/src/isgit.js

```diff
@@ -32,6 +33,12 @@ exports.gitInfo = (url) => {
       .replace('/tar.gz/', '.git#')
   }
   const info = gitinfo.fromUrl(url)
+  if (!info) {
+    throw new Error(`Invalid git url: ${url}`)
+  }
+  if (!Object.hasOwn(info, 'committish')) {
+    info.committish = null
+  }
```

Comment:
why wouldn't this suffice?

```suggestion
  info.committish ??= null
```

Reply:
This exists because error output is better if destructuring committish out of the object doesn't fail. 
If `undefined` was explicitly returned, I wanted to keep it to make it easier to figure out what's going on.